src/test.js

```diff
@@ -550,6 +550,13 @@ function test( testName, expected, callback, async ) {
 	if ( arguments.length === 2 ) {
 		callback = expected;
 		expected = null;
+	} else if ( arguments.length === 3 && typeof expected !== "function" ) {
+		if ( global.console && global.console.warn ) {
+			global.console.warn(
+				"'expected' argument of QUnit.test is deprecated. " +
+				"Call the expect() function explicitly instead."
```

Comment:
Thanks for the review.

>  like prefixing all the deprecation messages with an all caps DEPRECATED

Done.

> also the HTML Reporter might get this info from the logs and show it on the reporter messages as well.

I'm not sure my design is the best one, but I came up with the following:

* Add a QUnit.deprecationWarnings list
* In deprecateWarns helper (added in core/utilities.js), use console.warn if available and add warnings to QUnit.deprecationWarnings
* In HTML reporter, use QUnit.deprecationWarnings to show warning list and total count of warnings(e.g. Tests completed in 1092 milliseconds. 511 assertions of 513 passed, 2 failed.1 warnings)

Does that sounds good to you?

> Also, I would like to have an internal method for the warning messages instead of checking for console.warn everywhere.

Done. Added in core/utilities.js.